In [1]:
import numpy as np
import pandas as pd

# Задание 1   

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):   
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],   
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].   
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy   
Полученные значения должны быть равны.   
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,   
а затем с использованием функций из библиотек numpy и pandas.

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
zp.size

10

In [3]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
ks.size

10

In [4]:
n = 10

$$cov_{XY} = M(XY) - M(X)M(Y)$$

In [5]:
cov_ = (zp * ks).mean() - zp.mean() * ks.mean()
cov_

9157.839999999997

In [6]:
np.cov(zp, ks)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [7]:
cor_ = cov_ / (zp.std() * ks.std())
cor_

0.8874900920739158

In [8]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

# Задание 2   

Измерены значения IQ выборки студентов,    
обучающихся в местных технических вузах:   
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.   
Известно, что в генеральной совокупности IQ распределен нормально.   
Найдите доверительный интервал для математического ожидания с надежностью 0.95.   

In [9]:
IQ = pd.DataFrame([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [10]:
n = IQ.size
n

10

Найдем  𝑡𝜆  по таблице значений  𝑡𝜆=(𝜆,𝑛)  для  𝜆=0.95  и n = 10:   
$t_{\lambda} = 2.26$ 

$$\delta = \frac{t_{\lambda} \cdot \sigma}{\sqrt{n}}$$

In [11]:
t = 2.26
std = IQ.std(ddof=1)

In [12]:
delta = t * std / n ** 0.5 

In [13]:
M = IQ.mean()

In [14]:
T1 = M - delta
T1

0    110.563277
dtype: float64

In [15]:
T2 = M + delta
T2

0    125.636723
dtype: float64

In [16]:
def interval(M, t, std, n):
    delta = t * std / n ** 0.5
    T1 = M - delta
    T2 = M + delta
    return T1, T2

In [17]:
interval(M, t, std, n)

(0    110.563277
 dtype: float64, 0    125.636723
 dtype: float64)

Ответ: с надежностью 95% интервал (110.56; 125.64) покрывает математическое ожидание IQ студентов.

# Задание 3   

Известно, что рост футболистов в сборной распределен нормально   
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,   
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического    
ожидания с надежностью 0.95.

In [18]:
std = 25 ** 0.5
n = 27
M = 174.2
t = 2.052

In [19]:
interval(M, t, std, n)

(172.22546207937148, 176.1745379206285)

Ответ: с надежностью 95% интервал (172.22; 176.17) покрывает математическое ожидание роста футболистов.

# Задание 4

Выберите тему для проектной работы по курсу Теории вероятностей и математической статистики
и напишите ее в комментарии к Практическому заданию.

Тема проектной работы: "Исследование зависимости рейтинга фильмов от кассовых сборов и первоначальных инвестиций". 
    В маркетинге гипотеза может быть любой, например: между рейтингом и кассовыми сборами существует прямая зависимость. 
        Но в математической статистике гипотеза должна делать 
        предположение относительно распределения случайной величины или параметров ее распределения.
        Поэтому моя нулевая гипотеза состоит в том, что рейтинг имеет нормальное распределение, и кассовые сборы тоже 
        подчиняются закону нормального распределения.